punch/cli.py

```diff
@@ -152,21 +157,24 @@ def main(original_args=None):
     current_version = ver.Version.from_file(args.version_file, config.version)
     new_version = current_version.copy()
 
+    if args.part:
+        args.action = "punch:increase"
+        args.action_options = "part={}".format(args.part)
+
+    if args.set_part:
+        args.action = "punch:set"
+        args.action_options = args.set_part
+
     if args.action:
         action_dict = config.actions[args.action]
-        action = act.Action.from_dict(action_dict)
-        new_version = action.process_version(new_version)
-    else:
-        if args.part:
-            new_version.inc(args.part)
+        action_name = action_dict.pop('type')
 
-        if args.set_part:
-            if args.reset_on_set:
-                part, value = args.set_part.split('=')
-                new_version.set_and_reset(part, value)
-            else:
-                set_dict = dict(i.split('=') for i in args.set_part.split(','))
-                new_version.set(set_dict)
+        if args.action_options:
+            action_dict.update(hlp.optstr2dict(args.action_options))
+
+        action_class = ar.ActionRegister.get(action_name)
+        action = action_class(action_dict)
```

Comment:
`action_class` can be `None` because `action_name` is defined by user input; this will crash.

Reply:
:+1: